spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -829,6 +831,108 @@ When a repository method is called with a null value as an argument to one of it
 
 NOTE: The behavior of a query method when the method is called with a null argument is not defined by this specification, and is not portable between Jakarta Data providers.
 
+==== Restriction parameters
+
+The `Restriction<T>` type is a special parameter used to dynamically construct queries in a fluent and type-safe manner. Valid as parameters to methods that perform queries, such as methods annotated `@Find` or `@Delete`, restrictions allow the application to define additional conditions programmatically, often using a generated metamodel class (see <<metamodel-java-processor>>).
+
+When using `Restriction<T>` in repository methods:
+
+- The method *must* be annotated with either `@Find` or `@Query`. Without one of these annotations, the use of `Restriction<T>` is *not valid* and results in a `MappingException`.
+- When `@Find` or `@Query` is present, `Restriction<T>` can appear in *any position* in the parameter list. In these cases, the `Restriction<T>` provides the full query logic programmatically, and any query derivation from the method name is ignored.
+
+For example:
+
+[source,java]
+----
+@Query("WHERE (product.numAvailable > 0)")
+List<Product> search(Restriction<Product> restriction, Order<Product> order);
+
+@Find
+Page<Product> findAvailable(Restriction<Product> restriction, PageRequest pageRequest);
+----
+
+The following is *invalid* and will throw a `MappingException`:
```

Comment:
Now that you've decided Query by Method Name won't support Restriction, there are two possible errors that could be raised by the method.  MappingException for the lack of a parameter for the name.  UnsupportedOperationException for the attempt to supply Restriction to a method where it is not allowed. I don't think we can tell the Data Provider to choose one over the other. Either error would be valid.

```suggestion
The following is *invalid* because the method does not follow the Query by Method Name pattern of supplying a parameter per condition defined by the method name, nor does the method have an annotation that allows a `Restriction` parameter to be used.
```